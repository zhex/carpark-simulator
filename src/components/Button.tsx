import * as React from 'react';
import styled from 'styled-components';

const Button = styled.button`
    padding: 10px 20px;
    border-radius: 5px;
    border: 1px solid #999;
    background: white;
    color: black;
    cursor: pointer;
    outline: 0;

    &[disabled] {
        background: lightgrey;
        border-color: #999;
        color: #999;
    }

    &:active {
        background: #eee;
    }

    & + button {
        margin-left: 15px;
    }
`;

export const PrimaryButton = Button.extend`
    background: green;
    color: white;
    border-color: transparent;
    border-bottom-color: darkgreen;

    &:active {
        background: darkgreen;
    }
`;

export default Button;
