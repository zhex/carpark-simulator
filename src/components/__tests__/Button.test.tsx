import 'jest-styled-components';
import * as React from 'react';
import { create } from 'react-test-renderer';

import Button, { PrimaryButton } from '../Button';

it('Button Component', () => {
    const tree = create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Primary Button Component', () => {
    const tree = create(<PrimaryButton />).toJSON();
    expect(tree).toMatchSnapshot();
});
