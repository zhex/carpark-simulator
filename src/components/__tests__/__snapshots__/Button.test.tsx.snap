// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component 1`] = `
.c0 {
  padding: 10px 20px;
  border-radius: 5px;
  border: 1px solid #999;
  background: white;
  color: black;
  cursor: pointer;
  outline: 0;
}

.c0[disabled] {
  background: lightgrey;
  border-color: #999;
  color: #999;
}

.c0:active {
  background: #eee;
}

.c0 + button {
  margin-left: 15px;
}

<button
  className="c0"
/>
`;

exports[`Primary Button Component 1`] = `
.c0 {
  padding: 10px 20px;
  border-radius: 5px;
  border: 1px solid #999;
  background: white;
  color: black;
  cursor: pointer;
  outline: 0;
  background: green;
  color: white;
  border-color: transparent;
  border-bottom-color: darkgreen;
}

.c0[disabled] {
  background: lightgrey;
  border-color: #999;
  color: #999;
}

.c0:active {
  background: #eee;
}

.c0 + button {
  margin-left: 15px;
}

.c0:active {
  background: darkgreen;
}

<button
  className="c0"
/>
`;
